core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -103,7 +103,11 @@ class HistoryServer(
       // Note we don't use the UI retrieved from the cache; the cache loader above will register
       // the app's UI, and all we need to do is redirect the user to the same URI that was
       // requested, and the proper data should be served at that point.
-      res.sendRedirect(res.encodeRedirectURL(req.getRequestURI()))
+      if (req.getQueryString == null) {
```

Comment:
@srowen I think having an `if` statement like that makes it a little harder to read. How about the following, it has lesser repetition.
```
val requestURI = if (req.getQueryString == null) {
  req.getRequestURI
} else {
  req.getRequestURI + "?" + req.getQueryString
}
res.sendRedirect(res.encodeRedirectURL(requestURI))
```